service-base/src/main/java/org/eclipse/hono/service/HonoBeanNameGenerator.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service;
+
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanNameGenerator;
+
+/**
+ * Generates bean names that are identical to the class name.
+ */
+public class HonoBeanNameGenerator implements BeanNameGenerator {
```

Comment:
Thanks for doing the analysis and putting together this overview of options :+1: 

Option 1 seems to be easy to implement but it might be problematic to keep the bean name specified in the `@Component` annotation in sync with the class name (e.g. when refactoring the class).
Option 2 basically has the same problem as option 1
Option 3 seems to have the advantage that we can continue to use the class name as the bean name and still be resilient against changes to the component class names. It seems kind of awkward (and error prone), though, to require the definition of the name generator on the application class. When setting the name generator in this way, will it *overwrite* any bean names set explicitly, e.g. using `@Component("myBean")`? Is there any other way of setting the name generator in `AbstractApplication` so that we only need to define it once?